src/core/partition.h

```diff
@@ -56,9 +61,7 @@ typedef struct QUIC_CACHEALIGN QUIC_PARTITION {
     // Used for generating stateless retries.
     //
     CXPLAT_DISPATCH_LOCK StatelessRetryKeysLock;
-    CXPLAT_KEY* StatelessRetryKeys[2]; // Encryption of retry tokens
-    int64_t StatelessRetryKeysExpiration[2]; // Timestamp when the current stateless retry key expires.
-    BOOLEAN CurrentStatelessRetryKey; // Index into StatelessRetryKeys.
+    QUIC_RETRY_KEY StatelessRetryKeys[2];
```

Comment:
IMO, there is some non-trivial logic here (keeping only the key for the last two time intervals, access to the right one based on the timestamp). Some documentation would help.

```suggestion
    //
    // The current and previous stateless retry keys.
    // A new key and index are generated every `QUIC_STATELESS_RETRY_KEY_LIFETIME_MS`.
    // They are indexed based on the timestamp and their position on the array depends on the index parity.
    //
    QUIC_RETRY_KEY StatelessRetryKeys[2];
```

Reply:
I added more description to the abstract above.